CRM/Utils/Token.php

```diff
@@ -1956,4 +1911,44 @@ public static function getTokenDeprecations(): array {
     ];
   }
 
+  /**
+   * Get the tokens available for the domain.
+   *
+   * @param int $domainID
+   * @param bool $html
+   *
+   * @return array
+   * @throws \CRM_Core_Exception
+   */
+  protected static function getDomainTokens(int $domainID, bool $html): array {
+    $cacheKey = __CLASS__ . 'domain_tokens' . $html . '_' . $domainID . '_' . CRM_Core_I18n::getLocale();
+    if (!Civi::cache('metadata')->has($cacheKey)) {
+      if (CRM_Core_Config::domainID() === $domainID) {
+        $domain = CRM_Core_BAO_Domain::getDomain();
+      }
+      else {
+        $domain = new CRM_Core_BAO_Domain();
```

Comment:
honestly - we are being pretty coy not just loading it in the first place but maybe it will save a query...